File: 1365-how-many-numbers-are-smaller-than-the-current-number/1365-how-many-numbers-are-smaller-than-the-current-number.java
class Solution {
    public int[] smallerNumbersThanCurrent(int[] nums) {
        int  n = nums.length;
        int count = 0;
        int[] arr = new int[n];
        for (int i = 0;  i < n ; i++){
            for(int j = n-1; j >= 0; j--){
                if(nums[i]==nums[j] || i==j){
                    continue;
                }
                if(nums[i] > nums[j] ){
                    count++;
                }
                
            }
            arr[i] = count;
            count = 0;
        }
        return arr;
        
    }
}

File: 1365-how-many-numbers-are-smaller-than-the-current-number/1365-how-many-numbers-are-smaller-than-the-current-number.cpp
class Solution {
public:
    vector<int> smallerNumbersThanCurrent(vector<int>& nums) {
        vector<int> sorted  = nums;
        unordered_map<int,int> mp;

        sort(sorted.begin(),sorted.end());

        for(int i = 0; i<sorted.size(); i++){
            if(mp.count(sorted[i]) == 0){
                mp[sorted[i]] = i;
            }

        }

        vector<int > ans;

        for(int num : nums){
            ans.push_back(mp[num]);
        }
        return ans;

    }
};